tests/unit/nn/test_transforms.py

```diff
@@ -0,0 +1,157 @@
+def scaler(mean, scale):
+    scaler = StandardScaler()
+    scaler.mean_ = mean
+    scaler.scale_ = scale
+    return scaler
+
+
+@pytest.fixture
+def scale_transform(scaler, pad):
+    return ScaleTransform.from_standard_scaler(scaler, pad)
+
+
+@pytest.fixture
+def unscale_transform(scaler, pad):
+    return UnscaleTransform.from_standard_scaler(scaler, pad=0)
+
+
+@pytest.fixture
+def graph_transform(scale_transform):
+    return GraphTransform(V_transform=scale_transform, E_transform=scale_transform)
+
+
+@pytest.fixture
+def X():
+    return torch.tensor([[99.0, 99.0, 1.0, 2.0], [99.0, 99.0, 3.0, 4.0]])
+
+
+@pytest.fixture
+def prediction():
+    return torch.tensor([[1.0, 2.0]])
+
+
+@pytest.fixture
+def variance():
+    return torch.tensor([[0.1, 0.2]])
+
+
+@pytest.fixture
+def bmg():
+    V = torch.tensor([[99.0, 99.0, 1.0, 2.0], [99.0, 99.0, 3.0, 4.0]])
+    E = torch.tensor([[99.0, 99.0, 1.0, 2.0], [99.0, 99.0, 3.0, 4.0]])
+    return MockBatchMolGraph(V=V, E=E)
+
+
+def test_uneven_shapes():
+    with pytest.raises(ValueError):
+        ScaleTransform(mean=[0.0], scale=[1.0, 2.0])
+
+
+def test_padding(mean, scale, pad):
+    scale_transform = ScaleTransform(mean, scale, pad)
+    assert torch.all(scale_transform.mean[0, :pad] == 0.0).item()
+    assert torch.all(scale_transform.scale[0, :pad] == 1.0).item()
+
+
+def test_from_standard_scaler(mean, scale, scaler):
+    scale_transform = ScaleTransform.from_standard_scaler(scaler)
+
+    assert torch.all(scale_transform.mean == torch.tensor([0.0, 1.0])).item()
+    assert torch.all(scale_transform.scale == torch.tensor([2.0, 3.0])).item()
+
+
+def test_scale_transform_forward_train(scale_transform, X):
+    scale_transform.train()
+    output_X = scale_transform(X)
+    assert output_X is X
+
+
+def test_scale_transform_forward_eval(tensor_mean, tensor_scale, scale_transform, X):
+    scale_transform.eval()
+    output_X = scale_transform(X)
+    expected_X = (X - tensor_mean) / tensor_scale
+    assert torch.equal(output_X, expected_X)
+
+
+def test_unscale_transform_forward_train(unscale_transform, X):
+    unscale_transform.train()
+    output_X = unscale_transform(X)
+    assert output_X is X
+
+
+def test_unscale_transform_forward_eval(mean, scale, unscale_transform, prediction):
+    unscale_transform.eval()
+    output = unscale_transform(prediction)
+    expected = prediction * scale + mean
+    print(output, expected)
+    assert torch.equal(output, expected)
+
+
+def test_unscale_transform_variance_train(unscale_transform, variance):
+    unscale_transform.train()
+    output_variance = unscale_transform.transform_variance(variance)
+    assert output_variance is variance
+
+
+def test_unscale_transform_variance_eval(scale, unscale_transform, variance):
+    unscale_transform.eval()
+    output_variance = unscale_transform.transform_variance(variance)
+    expected_variance = variance * scale**2
+    assert torch.allclose(output_variance, expected_variance.float())
```

Comment:
The other tests use `torch.equal` but for some reason the second element in these two tensors differed by 1 byte so `torch.equal` returned `False`. 
`output_variance` is a single and `expected_variance` is a double, hence the `.float()`. Apparently `torch.equal` doesn't care about different float types because in `test_unscale_transform_forward_eval` the two tensors have different float types but `torch.equal` still returns `True`.

Reply:
Can we use `torch.testing.assert_close()`?

Reply:
Yes `torch.testing.assert_close(output_variance, expected_variance.float())` also works. Do you prefer that? Should I also change the `assert torch.equal`s?